packages/pyright-internal/src/pyrightFileSystem.ts

```diff
@@ -44,6 +55,53 @@ export class PyrightFileSystem implements FileSystem {
         return this._realFS.existsSync(this._getPath(path));
     }
 
+    fileExistsSync(path: string, canCache = false): boolean {
+        const splitPath = this._splitPath(path);
+
+        if (!canCache || !splitPath[0] || !splitPath[1]) {
+            if (!this.existsSync(path, canCache)) {
+                return false;
+            }
+            try {
+                const stats = this.statSync(path);
+                return stats.isFile();
+            } catch {
+                return false;
+            }
+        }
+
+        const entries = this.readdirEntriesSync(splitPath[0], /* canCache */ true);
+        const entry = entries.find((entry) => entry.name === splitPath[1]);
```

Comment:
can this be a problem if the directory has a lot of files?

Reply:
Possibly. That's one of the tradeoffs this change is trying to balance.

We're using the caching functionality primarily in contexts where we're already calling readdirEntriesSync within the import resolver (because we're scanning for native libraries whose names can't be generated statically), so I'm not that worried about it.